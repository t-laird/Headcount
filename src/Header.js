import React from 'react';

const Header = () => {
  return (
    <div className='Header'>
      <h1>HeadCount</h1>
    </div>
  )
}

export default Header;