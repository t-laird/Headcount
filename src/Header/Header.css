.Header {
  background: #ff4d4d;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center; }
  .Header h1 {
    margin: 0px;
    font-size: 90px; }
    .Header h1 span {
      border: 1px solid black;
      border-radius: 50%;
      font-size: 12px;
      padding: 5px;
      margin: 0px -30px 0px -25px;
      background-color: black;
      color: #ff4d4d; }

@media screen and (max-width: 650px) {
  .Header h1 {
    font-size: 60px; }
    .Header h1 span {
      font-size: 9px;
      margin: 0px -17px; } }

@media screen and (max-width: 450px) {
  .Header h1 {
    font-size: 45px; }
    .Header h1 span {
      font-size: 8px;
      margin: 0px -10px; } }
