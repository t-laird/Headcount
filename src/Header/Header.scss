// @import url('https://fonts.googleapis.com/css?family=Squada+One');

.Header {
  background-color: lighten(red, 15%);
  box-shadow: 5px 5px 5px 5px darkgrey;
  margin-bottom: 15px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  h1 {
    margin: 0px;
    font-size: 90px;
    // font-family: 'Squada One', sans-serif;
    // letter-spacing: .06em;
    // font-weight: 400; 
    span {
      border: 1px solid black;
      border-radius: 50%;
      font-size: 12px;
      padding: 5px;
      margin: 0px -30px 0px -25px;
      background-color: black;
      color: lighten(red, 15%);
      // letter-spacing: 0;
    }
  }

  .button-wrapper {
    display: flex;
    margin: 5px;
    button {
      border-radius: 50px;
      cursor: pointer;
      border: none;
      margin: 3px;
    }
  }
}

@media screen and (max-width: 650px) {
  .Header {
    h1 {
      font-size: 60px;
      span {
        font-size: 9px;
        margin: 0px -17px;
      }
    }
  }
}

@media screen and (max-width: 450px) {
  .Header {
    h1 {
      font-size: 45px;
      span {
        font-size: 8px;
        margin: 0px -10px;
      }
    }
  }
}