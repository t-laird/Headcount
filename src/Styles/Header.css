.Header {
  background-color: #ff4d4d;
  box-shadow: 5px 5px 5px 5px darkgrey;
  margin-bottom: 15px;
  display: flex;
  justify-content: center; }
  .Header h1 {
    margin: 10px 0px;
    font-size: 90px; }
    .Header h1 span {
      font-size: 12px;
      margin: 0px -25px; }

@media screen and (max-width: 650px) {
  .Header h1 {
    font-size: 60px; }
    .Header h1 span {
      font-size: 9px;
      margin: 0px -17px; } }

@media screen and (max-width: 450px) {
  .Header h1 {
    font-size: 45px; }
    .Header h1 span {
      font-size: 8px;
      margin: 0px -10px; } }
