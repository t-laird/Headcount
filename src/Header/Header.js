import React from 'react';
import Button from '../Button/Button.js';
import './Header.css';

import gradData from '../../data/high_school_graduation_rates.js';
import kinderData from '../../data/kindergartners_in_full_day_program.js';
import onlineEnrollment from '../../data/online_pupil_enrollment.js';
import householdIncome from '../../data/median_household_income.js';
import enrollment from '../../data/pupil_enrollment.js';
import remediationInHigherEducation from '../../data/remediation_in_higher_education.js';
import studentsInPoverty from '../../data/school_aged_children_in_poverty.js';
import specialEducation from '../../data/special_education.js';
import titleIstudents from '../../data/title_i_students.js';


const Header = (props) => {
  return (
    <div className='Header'>
      <h1>HeadCount 2 <span>dot</span> 0</h1>
      <div className='button-wrapper'>
        <Button
          type='gradData' 
          data={gradData}
          populateData={props.populateData}/>
        <Button
          type='kinderData' 
          data={kinderData}
          populateData={props.populateData}/>
        <Button
          type='onlineEnrollment' 
          data={onlineEnrollment}
          populateData={props.populateData}/>
        <Button
          type='householdIncome' 
          data={householdIncome}
          populateData={props.populateData}/>
        <Button
          type='enrollment' 
          data={enrollment}
          populateData={props.populateData}/>
        <Button
          type='remediationInHigherEducation' 
          data={remediationInHigherEducation}
          populateData={props.populateData}/>
        <Button
          type='studentsInPoverty' 
          data={studentsInPoverty}
          populateData={props.populateData}/>
        <Button
          type='specialEducation' 
          data={specialEducation}
          populateData={props.populateData}/>
        <Button
          type='titleIstudents' 
          data={titleIstudents}
          populateData={props.populateData}/>
      </div>
    </div>
  );
};

export default Header;