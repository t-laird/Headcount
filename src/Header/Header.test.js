import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { shallow, mount } from 'enzyme';

describe('Header test', () => {
  it('should render the correct components', () => {
    const renderedHeader = shallow(<Header cards={[]}/>);
    expect(renderedHeader.find('.Header').length).toEqual(1);
    expect(renderedHeader.find('h1').length).toEqual(1);
  })

  it('should match the Header snapshot', () => {
    const renderedHeader = shallow(<Header cards={[]}/>);
    expect(renderedHeader).toMatchSnapshot();
  })
});