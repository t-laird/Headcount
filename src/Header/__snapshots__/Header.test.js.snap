exports[`Header test should match the Header snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    cards={Array []} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        HeadCount 2 
        <span>
                dot
        </span>
         0
</h1>,
      "className": "Header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "HeadCount 2 ",
          <span>
            dot
</span>,
          " 0",
        ],
      },
      "ref": null,
      "rendered": Array [
        "HeadCount 2 ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "dot",
          },
          "ref": null,
          "rendered": "dot",
          "type": "span",
        },
        " 0",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          HeadCount 2 
          <span>
                    dot
          </span>
           0
</h1>,
        "className": "Header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "HeadCount 2 ",
            <span>
              dot
</span>,
            " 0",
          ],
        },
        "ref": null,
        "rendered": Array [
          "HeadCount 2 ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "dot",
            },
            "ref": null,
            "rendered": "dot",
            "type": "span",
          },
          " 0",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
